libs/s25main/mapGenerator/Harbors.cpp

```diff
@@ -68,16 +68,8 @@ namespace rttr { namespace mapGenerator {
         std::vector<std::vector<MapPoint>> coasts;
         std::set<MapPoint, MapPointLess> visited;
 
-        const auto partOfRiver = [&rivers](const MapPoint& pt) {
-            return helpers::contains_if(rivers, [&pt](const River& river) { return river.find(pt) != river.end(); });
-        };
-
-        const auto distanceToRiver = DistancesTo(map.size, partOfRiver);
-        const auto allWater = [&map](const MapPoint& pt) { return map.textureMap.All(pt, IsWater); };
-
-        const auto isCoast = [&map, allWater, &distanceToRiver](const MapPoint& pt) {
-            return map.textureMap.Any(pt, IsLand) && map.textureMap.Any(pt, IsWater)
-                   && helpers::contains_if(map.textures.GetNeighbours(pt), allWater) && distanceToRiver[pt] >= 5;
+        const auto isCoast = [&map](const MapPoint& pt) {
+            return map.textureMap.Any(pt, IsLand) && map.textureMap.Any(pt, IsWater);
```

Comment:
I'm unsure about the logic change here:
- prior: One of the neighbor points must be fully surrounded by water.
- now: any neighboring texture of this point must be water

Does this really work? I mean great if it does, as it's much faster, but can't this lead to problems?

Reply:
The "one of the neighbor points must be fully surrounded by water" was there to  ignore harbors being placed on 1-texture-rivers. Later I passed all rivers to the function anyway, so I could have removed that code before I suppose. Unless I'm missing some edge case I just forgot about ...

Reply:
There might be 2 islands very close to each other I guess. I wonder if there's a simpler solution to include that case.

Reply:
I guess this won't be trivial, but if you have an idea when this could break, why not add a testcase for it? "Just" prepare a map as-if it was random-generated so far with the situation you describe and call this function.
Then you can either really BOOST_TEST it, split the function so you can or at least debug it and verify that way that it works as expected.
Up to you, we can postpone this to when an actual bug shows up and if it doesn't then we can save the work.